src/context/MarketContext.js

```diff
@@ -87,178 +96,143 @@ const MarketContextProvider = ({ children }) => {
-                    new BigNumber(10).pow(item.decimals)
-                  );
-                }
+                isEnabled = toDecimalAmount(wallet.tokenAllowance).isGreaterThan(walletBalance);
               }
+            }
 
-              // supply balance
-              asset.supplyBalance = new BigNumber(supplyBalance).div(
-                new BigNumber(10).pow(item.decimals)
-              );
-
-              // borrow balance
-              asset.borrowBalance = new BigNumber(borrowBalance).div(
-                new BigNumber(10).pow(item.decimals)
-              );
-
-              // hypotheticalLiquidity
-              // return data type: (uint(err), liquidity, shortfall);
-              asset.hypotheticalLiquidity = account
-                ? await comptrollerContract.methods
-                    .getHypotheticalAccountLiquidity(
-                      account,
-                      asset.vtokenAddress,
-                      totalBalance,
-                      0
-                    )
-                    .call()
-                : ['0', '0', '0'];
-
-              const supplyBalanceUSD = asset.supplyBalance.times(
-                asset.tokenPrice
-              );
-              const borrowBalanceUSD = asset.borrowBalance.times(
-                asset.tokenPrice
-              );
-
-              totalBorrowBalance = totalBorrowBalance.plus(borrowBalanceUSD);
-              if (asset.collateral) {
-                totalBorrowLimit = totalBorrowLimit.plus(
-                  supplyBalanceUSD.times(asset.collateralFactor)
-                );
-              }
+            const asset = {
+              key: index,
+              id: item.id,
+              img: item.asset,
+              vimg: item.vasset,
+              name: market.underlyingSymbol || '',
+              symbol: market.underlyingSymbol || '',
+              decimals: item.decimals,
+              tokenAddress: market.underlyingAddress,
+              vsymbol: market.symbol,
+              vtokenAddress,
+              supplyApy: new BigNumber(market.supplyApy || 0),
+              borrowApy: new BigNumber(market.borrowApy || 0),
+              xvsSupplyApy: new BigNumber(market.supplyVenusApy || 0),
+              xvsBorrowApy: new BigNumber(market.borrowVenusApy || 0),
+              collateralFactor: new BigNumber(
+                market.collateralFactor || 0
+              ).div(1e18),
+              tokenPrice: new BigNumber(market.tokenPrice || 0),
+              liquidity: new BigNumber(market.liquidity || 0),
+              borrowCaps: new BigNumber(market.borrowCaps || 0),
+              totalBorrows: new BigNumber(market.totalBorrows2 || 0),
+              walletBalance,
+              supplyBalance,
+              borrowBalance,
+              isEnabled,
+              collateral,
+              percentOfLimit
+            };
+
+            return asset;
+          }
+        )
 
```

Comment:
just `return { .... }` will be good

Reply:
b7cd4500dccd10935ae75ecea28990aff8b15d83